src/fluree/db/ledger/consensus/tcp.clj

```diff
@@ -200,41 +200,57 @@
               (recur conn))))))))
 
 ;; store client event loop here to be shared across all client connections
-(def client-event-loop (atom nil))
+(def client-event-loops (atom {}))
 
 (defn get-client-event-loop
   "Will create a new client event loop if one doesn't already exist."
-  []
-  (or @client-event-loop
-      (let [new-state (swap! client-event-loop
-                             (fn [existing]
-                               (if existing
-                                 existing
-                                 (ntcp/event-loop))))]
-        new-state)))
+  [this-server]
+  (or (get @client-event-loops this-server)
+      (let [new-state (swap! client-event-loops
+                             (fn [loops]
+                               (if-let [existing (get loops this-server)]
+                                 loops
+                                 (assoc loops this-server (ntcp/event-loop)))))]
+        (get new-state this-server))))
 
 
 (defn shutdown-client-event-loop
   "Shuts down client event loop if it exists, returns true."
-  []
-  (swap! client-event-loop
-         (fn [cel]
-           (when cel
-             (ntcp/shutdown! cel))
-           nil))
+  [this-server]
+  (swap! client-event-loops
+         (fn [loops]
+           (if-let [cel (get loops this-server)]
+             (do (log/debug "Shutting down tcp client event loop" this-server)
+                 (ntcp/shutdown! cel)
+                 (dissoc loops this-server))
+             loops)))
   true)
 
+(defn validate-remote-address!
+  "Validates that we are given a server address that we can connect to. Will spin until
+  the address is resolvable to facilitate startup in environments that don't have a DNS
+  entry for a given hostname."
+  [host port]
+  (assert (string? host))
+  (assert (int? port))
+  (loop [retries 0]
+    (let [addr (InetSocketAddress. ^String host ^Integer port)]
+      (if (.isUnresolved addr)
+        (do (Thread/sleep 5000)
```

Comment:
Should we put the sleep after the log so we see it closer to the time it wasn't resolvable?